### PR TITLE
Update OP-TEE to the v3.18.0 release

### PR DESCRIPTION
**Pull request by ricardosalveti**
Update OP-TEE related recipes to the v3.18.0 release, including our updated 3.18.0 optee-os fork.

### CODE REVIEW

**Review by quaresmajose — APPROVED** [review]
LGTM, I have some fixed for op-tee in main-next which maybe no longer exist in this version.

**Comment by ricardosalveti** [comment]
Tested on kv260 and uz3eg-iocc-sec, but failed on imx8mm-evk-sec (still investigating with upstream).

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by angolini — APPROVED** [review]

**Comment by ricardosalveti** [comment]
After bumping am's BSP (tested by Tim):

am62 xtest
```
+-----------------------------------------------------
29994 subtests of which 0 failed
123 test cases of which 0 failed
0 test cases were skipped
TEE test application done!
```
am64 xtest
```
+-----------------------------------------------------
30051 subtests of which 0 failed
123 test cases of which 0 failed
0 test cases were skipped
TEE test application done!
```

**Comment by ricardosalveti** [comment]
imx8qm-mek (tested by Igor):
```
+-----------------------------------------------------
29999 subtests of which 0 failed
123 test cases of which 0 failed
0 test cases were skipped
TEE test application done!
```

File: meta-lmp-base/recipes-security/optee/optee-os-fio.inc [hunk 1]
```diff
@@ -21,7 +21,7 @@ CVE_PRODUCT = "linaro:op-tee op-tee:op-tee_os"
 PROVIDES = "virtual/optee-os"
 
 SRCBRANCH ?= "master"
-SRC_URI = "git://github.com/foundriesio/optee_os.git;protocol=https;branch=${SRCBRANCH}"
+SRC_URI = "git://github.com/ricardosalveti/optee_os.git;protocol=https;branch=${SRCBRANCH}"
```
Comment (Tim-Anderson) [inline meta-lmp-base/recipes-security/optee/optee-os-fio.inc:24]: did you want this coming from your repo?

**Review by Tim-Anderson — COMMENTED** [review]
just the one question the reset LGTM

File: meta-lmp-base/recipes-security/optee/optee-os-fio.inc (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-security/optee/optee-os-fio.inc:24]: I will update once the final patch set is tested with this PR (still got a couple more patches to add).

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
I'm also adding the related versal updates (still tracking my own trees, but will update once done with testing and integration).

**Comment by ldts** [comment]
**Notes Versal**

- failed on 4005.7 due to unaligned request (expected)
       _regression_4005.7 AE case 6 algo 0x40000810 line 2430_
- skipped test 1008, never ending (new)

**xtest -t regression -x 1008**
```
+-----------------------------------------------------
25414 subtests of which 2 failed
97 test cases of which 1 failed
1 test case was skipped
TEE test application done!
```
**xtest -t pkcs11**
```
+-----------------------------------------------------
3765 subtests of which 0 failed
25 test cases of which 0 failed
0 test cases were skipped
TEE test application done!
```

**Comment by ricardosalveti** [comment]
8mp (tested by Vanessa):
```
+-----------------------------------------------------
29899 subtests of which 0 failed
122 test cases of which 0 failed
0 test cases were skipped
TEE test application done!
```

**Pull request merged**